packages/product-json-to-csv/src/writer.js

```diff
@@ -0,0 +1,101 @@
+import archiver from 'archiver'
+import EmitOnce from 'single-emit'
+import fs from 'fs'
+import json2csv from 'json2csv'
+import path from 'path'
+import slugify from 'slugify'
+import tmp from 'tmp'
+import mapHeaders from './map-headers'
+
+// Accept a highland stream and write the output to a single file
+export function writeToSingleCsvFile(productStream, output, logger, headers) {
+  const trimmedHeaders = headers.map(header => header.trim())
+  output.write(`${trimmedHeaders.join(',')}\n`) // Write headers first
+  const columnNames = mapHeaders(trimmedHeaders)
+  productStream
+    .each(product => {
+      const csvData = json2csv({
+        data: product,
+        fields: columnNames,
+        hasCSVColumnTitle: false,
+        quotes: '',
+      })
+      output.write(`${csvData}\n`)
```

Comment:
How does it know when its done without returning anything?

Reply:
One of the beauties of `highland`
`output.end()` is called in the `.done()` method of the highland stream which closes it